CUDLR/Scripts/Console.cs

```diff
@@ -95,9 +116,23 @@ public class Console {
 
     /* Logs string to output */
     public static void Log(string str) {
-      Instance.m_output.Add(str);
-      if (Instance.m_output.Count > MAX_LINES) 
-        Instance.m_output.RemoveAt(0);
+        Log(str, "");
+    }
+
+    /* Logs string to output with specified color corresponding to the message color name */
+    public static void Log(string str, string colorName)
+    {
+        if((str != "") && (Instance.m_messageColors.ContainsKey(colorName)))
```

Comment:
Not sure why you're checking for empty string here... but a better idea might be to probably wrap the entire function in String.IsNullOrEmpty(str)